skymarshal/skycmd/cf_flags.go

```diff
@@ -81,7 +87,20 @@ func (flag *CFTeamFlags) GetUsers() []string {
 func (flag *CFTeamFlags) GetGroups() []string {
 	var groups []string
 	groups = append(groups, flag.Orgs...)
-	groups = append(groups, flag.Spaces...)
+	groups = append(groups, flag.SpacesAll...)
+	for _, space := range flag.Spaces {
+		groups = append(groups, fmt.Sprintf("%s:developer", space))
+	}
+	for _, space := range flag.SpaceDevelopers {
+		groups = append(groups, fmt.Sprintf("%s:developer", space))
+	}
+	for _, space := range flag.SpaceAuditors {
+		groups = append(groups, fmt.Sprintf("%s:auditor", space))
+	}
+	for _, space := range flag.SpaceManagers {
+		groups = append(groups, fmt.Sprintf("%s:manager", space))
+	}
```

Comment:
Something makes me uneasy about this method of serializing these groups. It looks like spaces can have a colon in their name (based on [this regex](https://github.com/cloudfoundry/cloud_controller_ng/blob/0fc14205ce043c1f22a398c67ad080326f9ad115/app/models/runtime/space.rb#L10) and the fact that Ruby's `[[:punct:]]` pattern matches colons). Is there anything bad that can happen here if my space name has a colon in it?

Reply:
Nothing bad. If dex returns a group like `some-org:some-space-has-c::olon` [here](https://github.com/concourse/dex/blob/8bbc2a6df1eaa27aa67e54c7948002c709ed5e0e/connector/cf/cf.go#L292), as long as in the team auth config the same string is set by `--cf-sapce-developer some-org:some-space-has-c::olon`, everything should work. We don't have restriction on team auth configuration.